test/__tests__/warehouse.test.js

```diff
@@ -1233,4 +1233,119 @@ describe("isBlank", () => {
       expect(isBlank(testCase.input)).toEqual(testCase.expected);
     });
   }
-});
\ No newline at end of file
+});
+
+describe("context/group traits", () => {
+  const testCases = [
```

Comment:
[minor] It might be a bit more readable if we separate the tests for group events.

Reply:
Done